src/wxSimpleJSON.cpp

```diff
@@ -148,6 +148,20 @@ wxArrayString wxSimpleJSON::GetValueArrayString(const wxMBConv &conv) const
     return arr;
 }
 
+std::vector<wxString> wxSimpleJSON::GetValueStringVector(const wxMBConv& conv) const
```

Comment:
Why do we need vector version? Whats is wrong with `wxArrayString wxSimpleJSON::GetValueArrayString(const wxMBConv &conv) ` ?

Reply:
I get the vague impression withing the community that the wx container classes are being soft deprecated in favor of regular STL ones.  At least for me, I only use `std::vector<wxString>` instead of things like `wxArrayString`, so this seemed like a nice convenience function for those situations.

Reply:
So, I added `#ifdef wxUSE_STL` guards instead of removing the functions entirely.  I don't mean to be difficult, but I really prefer to keep vector helper functions in here as an alternative to wx containers.  It just seem easier to work with vectors when integrating with generic functions and other libraries.

Reply:
In that case ifdef isn't needed because `wxUSE_STL` means internally wx used either its own containers or std:: ones.
Am still against std::vector ones because we can extend the same reasoning to wxString. Why wxString and not `std::string`?

Do you have use case where wxArrayString does not work?

Reply:
Interestingly, they do say that you should try to use `std::wstring` instead of `wxString`:

https://docs.wxwidgets.org/trunk/classwx_string.html

I don't exactly agree with that though.  I would prefer `std::wstring`, but its lack of easy case-insensitive comparison and code-page conversion makes it difficult to work with. In this regard, I do prefer the wx specific `wxString`.  So I guess I'm hypocritical in this regard.  IDK, I just never use the wx containers, they seem like relics from pre-1998 when STL support was spotty.  At this point I see nothing that they offer over standard containers, as opposed to wxString that does add important features lacking in standard strings.  Using wx containers just seem like a confusing and difficult API to work with when `std::vector` is the de facto container in most new code these days.  Just my opinion though.

Reply:
>Interestingly, they do say that you should try to use std::wstring instead of wxString:
You are right, my point was to show that with the same logic we can go for `std::` version of string class instead of `wx` version.

> I just never use the wx containers, they seem like relics from pre-1998 when STL support was spotty
But wxArrayString  according to the docs supports full vector API, and with wxUSE_STL=1 it uses std containers. So I still think, wxArrayString should be preferred over std::vector as I don't see extra advantage. With those two action wxArrayString becomes an  std::vector wth extra methods, which I think is great. All of std::vector plus wxArrayString specific methods

Here is the quote for completion sake:
>It also has the full set of std::vector<wxString> compatible methods, including nested iterator and const_iterator classes which should be used in the new code for forward compatibility with the future wxWidgets versions.

With these pointed out, do you still think std::vector methods is a way to go?

>Just my opinion though.
Always different opinions are welcome. Discussions produces better projects. Even more, different opinions, when well handled produces the best of the projects!

Reply:
Personally, I think so.  I'm using `std::vector` to return boolean arrays and something else in here also, and I think that looks better API wise than using `wxVector`.  IDK, it just feels more modern I suppose...it just looks weird and clunky to me returning something like `wxVector<bool>`.

Reply:
Well since there is no consensus here, and the methods does not hurt, I will accept them.
If someone comes with strong argument, we can remove them if need be before next release